### PR TITLE
[PHP83] [MAD-14862] nxCLI updates for PHP83

### PR DESCRIPTION
**Pull request by davemednick**

### CODE REVIEW

File: src/DumpCommand.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $process = Process::fromShellCommandline($command);
         } else {
             // old Symfony way
-            $process = new Process($command);
+            $process = new Process([$command]);
```
Comment (nortoh) [inline src/DumpCommand.php]: I believe you need to explode the command by a space and pass it in as:

Example
```php
new Process(['ls', '-al']);
```
https://symfony.com/doc/current/components/process.html#usage

File: src/DumpCommand.php [hunk 2]
```diff
@@ -81,12 +81,13 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         // find all commands
         $command = $script . ' list ' . $scriptOptions . ' --format=xml';
+        $command_exploded = [$script . ' list ' . $scriptOptions, '--format=xml'];
```
Comment (nortoh) [inline src/DumpCommand.php]: This is still built as a single string but now it's in an array. We need each part of the command as an array element

File: composer.json [hunk 3]
```diff
@@ -5,8 +5,8 @@
     "require": {
         "php": "^7.1|^8.0",
         "ext-simplexml": "*",
-        "symfony/console": "^2.5|^3|^4|^5|^6",
-        "symfony/process": "^2.5|^3|^4|^5|^6"
+        "symfony/console": "^2.5|^3|^4|^5|^6|^7",
+        "symfony/process": "^2.5|^3|^4|^5|^6|^7"
```
Comment (nortoh) [inline composer.json:9]: Don't forget we need a `replace` to use our fork

Example
```json
    "replace": {
        "original_vendor/original_name": "*"
    },
```

**Review by nortoh — CHANGES_REQUESTED** [review]

File: src/DumpCommand.php (hunk 2 above)
Comment (davemednick) [reply src/DumpCommand.php]: patience. lol. I was fixing as you were revieweing

**Review by nortoh — APPROVED** [review]

**Review by mpedrummer — APPROVED** [review]

File: src/DumpCommand.php [hunk 4]
```diff
@@ -86,7 +86,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $process = Process::fromShellCommandline($command);
         } else {
             // old Symfony way
-            $process = new Process($command);
+            $process = new Process(explode(' ', $command));
```
Comment (mclorch) [inline src/DumpCommand.php:89]: did process use to take string as arg and now it is taking ...stings or similar?

Comment (davemednick) [reply src/DumpCommand.php:89]: Yes - old Symfony's `Process` method took an array of strings and combined them to make the command.

**Pull request merged**